bindings/java/src/main/com/apple/foundationdb/tuple/TupleUtil.java

```diff
@@ -539,10 +539,10 @@ static int compareItems(Object item1, Object item2) {
 	}
 
 	static List<Object> unpack(byte[] bytes, int start, int length) {
-		List<Object> items = new LinkedList<Object>();
+		List<Object> items = new LinkedList<>();
 		int pos = start;
 		int end = start + length;
-		while(pos < bytes.length) {
+		while(pos < end) {
```

Comment:
What happens if `end` is past the end of the `bytes` array?

Reply:
In that case, you will get an `ArrayIndexOutOfBoundsException` on line 351 of `TupleUtil` when it tries do decode the value out of range: https://github.com/apple/foundationdb/blob/a799d2a08c9b3d9227f67adfd74a210cc4176c89/bindings/java/src/main/com/apple/foundationdb/tuple/TupleUtil.java#L351

Arguably, the correct thing is for this method (unpack) to do bounds checking early and then exit right then. I could add that, if we want.

Reply:
What does the rest of the Tuple layer do? If it behaves similarly to this, we can leave it and address it later as a whole if we want.

Reply:
I believe it generally does bounds checking at the last possible moment (i.e., when it gets the `ArrayIndexOutOfBoundsException`. That leads to another problem, which is that the decode logic for, say, floats will not necessarily do the right thing. For example:

```java
Tuple.fromBytes(Tuple.from(3.14f).pack(), 0, 3)
```

will return `(3.14f)` rather than throwing an error (because it doesn't check that the item is entirely located within the correct range). I think all of the fixed length types are like that, but the variable length things will stop before hitting their terminator.

Reply:
Ok, as we discussed a bit offline, I think we'll go ahead and merge this. If we want, we can raise a new issue to address the concern you just raised.